CRM/Civicase/Hook/PostProcess/SaveCaseCategoryInstance.php

```diff
@@ -21,25 +21,23 @@ public function run($formName, CRM_Core_Form $form) {
     $caseCategoryValues = $form->getVar('_submitValues');
     $instanceTypeValue = $caseCategoryValues[self::INSTANCE_TYPE_FIELD_NAME];
     $caseCategoryValue = $caseCategoryValues['value'];
-    $this->saveCaseCategoryInstance($caseCategoryValue, $instanceTypeValue);
+
+    $this->saveCaseCategoryInstance([
+      'category_id' => $caseCategoryValue,
+      'instance_id' => $instanceTypeValue,
+      'singular_label' => $caseCategoryValues['case_category_singular_label'],
+    ]);
   }
```

Comment:
Same comment applies as in AddCaseCategoryInstanceField.php, we should add a new class to handle this or update the class name to reflect what it does.